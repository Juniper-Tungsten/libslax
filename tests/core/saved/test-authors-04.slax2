/* Machine Crafted with Care (tm) by slaxWriter */
version 1.1;

param $name = "Poe";
var $favorites := {
    <name hidden="yes"> "Spillane";
    <name> "Doyle";
}

match / {
    <out> {
        /* Parameters are passed by name */
        call test($elt = "author", $name);
    }
}

template test ($name, $elt = "default") {
    for $this ($favorites/name) {
        if ($name == $this && not(@hidden)) {
            element $elt {
                copy-of .//author[name/last == $this];
            }
        
        } else if ($name == .) {
            message "Hidden: " _ $name;
        }
    }
    
}
