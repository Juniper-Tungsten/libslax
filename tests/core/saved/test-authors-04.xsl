<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" version="1.0" extension-element-prefixes="slax-ext">
  <xsl:param name="name" select="&quot;Poe&quot;"/>
  <xsl:variable name="favorites-temp-1">
    <name hidden="yes">Spillane</name>
    <name>Doyle</name>
  </xsl:variable>
  <xsl:variable xmlns:slax-ext="http://xmlsoft.org/XSLT/namespace" name="favorites" select="slax-ext:node-set($favorites-temp-1)"/>
  <xsl:template match="/">
    <out>
      <!-- Parameters are passed by name -->
      <xsl:call-template name="test">
        <xsl:with-param name="elt" select="&quot;author&quot;"/>
        <xsl:with-param name="name" select="$name"/>
      </xsl:call-template>
    </out>
  </xsl:template>
  <xsl:template name="test">
    <xsl:param name="name"/>
    <xsl:param name="elt" select="&quot;default&quot;"/>
    <xsl:variable name="slax-dot-1" select="."/>
    <xsl:for-each select="$favorites/name">
      <xsl:variable name="this" select="."/>
      <xsl:for-each select="$slax-dot-1">
        <xsl:choose>
          <xsl:when test="$name = $this and not(@hidden)">
            <xsl:element name="{$elt}">
              <xsl:copy-of select=".//author[name/last = $this]"/>
            </xsl:element>
          </xsl:when>
          <xsl:when test="$name = .">
            <xsl:message>
              <xsl:value-of select="concat(&quot;Hidden: &quot;, $name)"/>
            </xsl:message>
          </xsl:when>
        </xsl:choose>
      </xsl:for-each>
    </xsl:for-each>
  </xsl:template>
</xsl:stylesheet>
