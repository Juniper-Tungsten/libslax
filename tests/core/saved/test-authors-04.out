<?xml version="1.0"?>
<out>
  <author>
    <author>
      <name>
        <first>Edgar Allan</first>
        <last>Poe</last>
      </name>
      <life-span>
        <born>1809</born>
        <died>1849</died>
      </life-span>
    </author>
  </author>
</out>
